### ISSUE TITLE
Uniprot data to ingest

### ISSUE THREAD

**Issue by pombase-admin** [pombase/pombase-chado#52]
Most of the information in UniProt is already in GeneDB but 
Uniprot may have information about catalytic activities and residues which we could potentially load into Chado

Original comment by: ValWood

**Comment by ValWood** [comment]
Possible data to ingest

1. synonyms
2. Modification data
3. Variation
4. Rhea  reactions

**Comment by ValWood** [comment]
Also mentioned here:https://github.com/pombase/pombase-chado/issues/1126

**Comment by kimrutherford** [comment]
Looks like we can get a lot of the information we need for this issue and for pombase/website#2115 and pombase/pombase-chado#1172 using the UniProt search API.  This URL gets us a gzipped TSV file with a row for each pombe entry and with columns for  catalytic activity, signal peptide, active sites, binding sites, synonyms, post-translational modification and modified residues:

https://rest.uniprot.org/uniprotkb/stream?compressed=true&fields=accession%2Cft_signal%2Cft_transit%2Cxref_pombase%2Cft_binding%2Cft_act_site%2Ccc_catalytic_activity%2Cgene_synonym%2Ccc_ptm%2Cft_mod_res&format=tsv&query=%28%28organism_id%3A284812%29%29

There's a little bit of parsing involved to get what we need from the TSV file but it's miles easier than parsing the full XML entries.

I've added the current results of the query above to SVN:
```
pombe-embl/external_data/uniprot_data_from_api.tsv
```

There might be other useful information we can grab like this.

The URL is created by clicking the download button on a multi-protein result page, like:
https://www.uniprot.org/uniprotkb?query=%28organism_id%3A284812%29

Then change the format to TSV and select columns.  There's a "Generate URL for API" link at the bottom.

**Comment by ValWood** [comment]
OK.
We need to map the modifications to PRO. Me or @PCarme can do that if you provide a list of terms.
Then we need to filter the exact duplicates (these should all be supported by a publication)

I

**Comment by ValWood** [comment]
I think we will have all of the catalytic activities already via GO IEA to Rhea

**Comment by kimrutherford** [comment]
In case anything looks interesting, here are the options for extra fields when downloading from the UniProt advanced search:

 - Names & Taxonomy
   - Entry Name
   - Gene Names
   - Gene Names (ordered locus)
   - Gene Names (ORF)
   - Gene Names (primary)
   - Gene Names (synonym)
   - Organism
   - Organism (ID)
   - Protein names
   - Proteomes
   - Taxonomic lineage
   - Taxonomic lineage (Ids)
   - Virus hosts
 - Sequences
   - Alternative products (isoforms)
   - Alternative sequence
   - Erroneous gene model prediction
   - Fragment
   - Gene encoded by
   - Length
   - Mass
   - Mass spectrometry
   - Natural variant
   - Non-adjacent residues
   - Non-standard residue
   - Non-terminal residue
   - Polymorphism
   - RNA Editing
   - Sequence
   - Sequence caution
   - Sequence conflict
   - Sequence uncertainty
   - Sequence version
 - Function
   - Absorption
   - Active site
   - Binding site
   - Catalytic activity
   - Cofactor
   - DNA binding
   - EC number
   - Activity regulation
   - Function [CC]
   - Kinetics
   - Pathway
   - pH dependence
   - Redox potential
   - Rhea ID
   - Site
   - Temperature dependence
 - Miscellaneous
   - Annotation
   - Caution
   - Keywords
   - Keyword ID
   - Miscellaneous [CC]
   - Protein existence
   - Reviewed
   - Tools
   - UniParc
   - Comments
   - Features
 - Interaction
   - Interacts with
   - Subunit structure
 - Expression
   - Developmental stage
   - Induction
   - Tissue specificity
 - Gene Ontology (GO)
   - Gene Ontology (biological process)
   - Gene Ontology (cellular component)
   - Gene Ontology (GO)
   - Gene Ontology (molecular function)
   - Gene Ontology IDs
 - Pathology & Biotech
   - Allergenic Properties
   - Biotechnological use
   - Disruption phenotype
   - Involvement in disease
   - Mutagenesis
   - Pharmaceutical use
   - Toxic dose
 - Subcellular location
   - Intramembrane
   - Subcellular location [CC]
   - Topological domain
   - Transmembrane
 - PTM / Processing
   - Chain
   - Cross-link
   - Disulfide bond
   - Glycosylation
   - Initiator methionine
   - Lipidation
   - Modified residue
   - Peptide
   - Post-translational modification
   - Propeptide
   - Signal peptide
   - Transit peptide
 - Structure
   - 3D
   - Beta strand
   - Helix
   - Turn
 - Publications
   - PubMed ID
   - DOI ID
 - Date of
   - Date of creation
   - Date of last modification
   - Date of last sequence modification
   - Entry version
 - Family & Domains
   - Coiled coil
   - Compositional bias
   - Domain [CC]
   - Domain [FT]
   - Motif
   - Protein families
   - Region
   - Repeat
   - Sequence similarities
   - Zinc finger

**Comment by ValWood** [comment]
- Virus hosts SHOULD HAVE NO DATA
 - Sequences
   - Alternative products (isoforms) A LIST OF PROTEINS WITH ISOFORMS WOULD BE USEFUL TO CHECK BUT ISN'T AN AREA OF FOCUS FOR US RIGHT NOW


 - Function
   - Active site DISPLAY IN PROTEIN VIEWER
   - Binding site DISPLAY IN PROTEIN VIEWER
 
   - Cofactor  A LIST WOULD BE USEFUL BUT I THINK WE ARE COVERED BY GO



   - Kinetics  WE COULD DISPLAY THIS



 
 - Interaction WE CAN LOOK AT THESE LATER BUT SHOULD BE COVERED
   - Interacts with
   - Subunit structure 

  
 - PTM / Processing (I CAN PROVIDE A MAPPING FOR THESE BUT WE ONLY NEED TO REPORT WHAT WE DON"T ALREADY HAVE)
   - Cross-link
   - Disulfide bond
   - Glycosylation
   - Initiator methionine
   - Lipidation
   - Modified residue
   - Peptide
   - Post-translational modification
   - Propeptide
   - Signal peptide
   - Transit peptide

   - Beta strand THESE WOULD BE USEFUL IN THE PROTEIN VIEWER, WE WERE GOING TO GET THEM FROM PDB
   - Helix
   - Turn
   - Compositional bias

**Comment by kimrutherford** [comment]
Thanks Val.

Here's the update API URL, adding the extra fields:
https://rest.uniprot.org/uniprotkb/stream?compressed=true&fields=accession%2Cft_signal%2Cft_transit%2Cxref_pombase%2Cft_binding%2Cft_act_site%2Ccc_catalytic_activity%2Cgene_synonym%2Ccc_ptm%2Cft_mod_res%2Ccc_cofactor%2Ckinetics&format=tsv&query=%28%28organism_id%3A284812%29%29

and the commands to update the data file in SVN:

```sh
cd pombe-embl
curl 'https://rest.uniprot.org/uniprotkb/stream?compressed=true&fields=accession%2Cft_signal%2Cft_transit%2Cxref_pombase%2Cft_binding%2Cft_act_site%2Ccc_catalytic_activity%2Cgene_synonym%2Ccc_ptm%2Cft_mod_res%2Ccc_cofactor%2Ckinetics&format=tsv&query=%28%28organism_id%3A284812%29%29' | gzip -d > external_data/uniprot_data_from_api.tsv
svn commit external_data/uniprot_data_from_api.tsv
```

**Comment by kimrutherford** [comment]
Updated API URL, including "Beta strand":

https://rest.uniprot.org/uniprotkb/stream?compressed=true&fields=accession%2Cid%2Cft_act_site%2Cft_binding%2Ccc_catalytic_activity%2Ccc_cofactor%2Ckinetics%2Cft_transit%2Cft_signal%2Ccc_ptm%2Cft_mod_res%2Cxref_pombase%2Cft_strand&format=tsv&query=%28%28organism_id%3A284812%29%29

**Comment by kimrutherford** [comment]
> Beta strand THESE WOULD BE USEFUL IN THE PROTEIN VIEWER, WE WERE GOING TO GET THEM FROM PDB

Prototype with Beta strands:
https://desktop.kmr.nz/gene_protein_features/SPBC1105.02c

Does that look OK?  We can workshop the colours.  :-)

![image](https://github.com/user-attachments/assets/bcf966d6-2607-457b-98a0-78fe2fc99908)

**Comment by ValWood** [comment]
looks good!

**Comment by ValWood** [comment]
and helix?

**Comment by ValWood** [comment]
Lys4 transit pepetide looks weird (base 1), is that because it has no coordinates? if so we shouldn't display for now (hopefully we will get if from other methods)

**Comment by kimrutherford** [comment]
> Lys4 transit pepetide looks weird (base 1), is that because it has no coordinates? 

It snuck in.  I'll remove it for tonight's load.  It's annotated as "TRANSIT 1..?" which is suppose to be filtered.

**Comment by kimrutherford** [comment]
> and helix?

How's this?:

https://desktop.kmr.nz/gene_protein_features/SPBC1105.02c

![image](https://github.com/user-attachments/assets/9da180d8-46b8-4531-8d77-8ffc0e23aba7)

**Comment by ValWood** [comment]
Looks good.

I think the labels should say "binding sites imported from uniprot"  etc.
Some of these things are curated by UniProt and some are automated by external packages.

**Comment by ValWood** [comment]
I suddenly had a thought. Many of the uniprot proten sequences don't match our updated ones.  To map many features correctly  you would need to run through Manu's code to correct the positions....

**Comment by kimrutherford** [comment]
> To map many features correctly you would need to run through Manu's code to correct the positions....

Unfortunately Manu's code won't help because we don't have the amino acids in the annotations.  So Manu's code can't search for the corrected coords.

**Comment by kimrutherford** [comment]
> I think the labels should say "binding sites imported from uniprot" etc.

Which labels?  There are labels for the tracks and labels on each feature.

We do mention UniProt in the track label mouse-over:

![image](https://github.com/user-attachments/assets/f117a097-13ef-4e83-adf7-222a4da18648)

**Comment by kimrutherford** [comment]
>  Many of the uniprot proten sequences don't match our updated ones.

So I guess we know it's safe to show the features for genes where our protein sequence matches UniProt's sequence?

**Comment by ValWood** [comment]
> So I guess we know it's safe to show the features for genes where our protein sequence matches UniProt's sequence?

Yes, I was thinking that. We will lose a lot but when the sequence updates filter through to UniProtKB  they will come through.
That means we will need to update, but probably once a year is enough. It will take a long time for the fixed because I think they need to be done manually (I kid you not)

**Comment by kimrutherford** [comment]
Can we trust the coordinates of the signal peptides and transit peptides if the sequence changes?  Are they always at the start of the sequence?

**Comment by ValWood** [comment]
>Can we trust the coordinates of the signal peptides and transit peptides if the sequence changes? Are they always at the start of the sequence?

good point, I suspect that there will not be many sequence changes for the ones with signal peptide/transit peptide.
if anything, they would not have been predicted because the N-term was incorrect. 
This means that the number should not really change, but we will pick up more as the sequence  corrections filter through.


Also I am wondering about active sites. For those we do have residue information right ? (but again, I suspect most of these would not change because they are highly expressed, highly conserved proteins.  Most of the changes were on proteins that are more divergent where the N-term residue was harder to predict.

The main problem would be the Alpha helix/beta sheet

**Comment by kimrutherford** [comment]
> We will lose a lot 

I've just run a check comparing the PomBase and UniProt protein sequences.  I only found 156 that differ.  Sounds like you were expecting more?

Here's the list:
https://www.pombase.org/results/from/id/7e87607d-9135-4df2-9778-c5da631d0dfe

**Comment by kimrutherford** [comment]
> Also I am wondering about active sites. For those we do have residue information right ?

Unfortunately we don't get the amino acid, just the position from the UniProt download.

**Comment by kimrutherford** [comment]
> good point, I suspect that there will not be many sequence changes for the ones with signal peptide/transit peptide.

I found these where there's a AA difference in the first 30 amino acids and there's a signal_peptide or transit_peptide:

SPAC13C5.03 signal_peptide
SPAC167.04 transit_peptide
SPAC1B3.18c transit_peptide
SPAC227.11c signal_peptide
SPAC23C11.17 transit_peptide
SPAC5H10.06c transit_peptide
SPAPB24D3.10c signal_peptide

**Comment by ValWood** [comment]
I thought it was more but we did a resubmission in 2017 ish so maybe it is only 156 since then. That's good!

**Comment by ValWood** [comment]
>Which labels? There are labels for the tracks and labels on each feature.

instead of "binding sites from UniProt"
I think we should say "binding sites imported from UniProt"
(for the binding sites  UniProt might curate them all, but for others UniProt might not be the originating source).
I think saying imported from implies better that we don't know who created them, or with what.
I will ask Antonia how UniProt makes/provides there so we can add it to the docs.

**Comment by kimrutherford** [comment]
OK, thanks.  I've changed the text.  The main site will have the change in 30 minutes  or so.

**Comment by kimrutherford** [comment]
> I found these where there's a AA difference in the first 30 amino acids and there's a signal_peptide or transit_peptide:
>
> SPAC13C5.03 signal_peptide
> SPAC227.11c signal_peptide
> SPAPB24D3.10c signal_peptide

Pascal gave me the motivation to try these three with SignalP.  It predicts one no, one maybe and one yes:
 
```
# SignalP-6.0	Organism: Eukarya	Timestamp: 20240803051334
# ID	Prediction	OTHER	SP(Sec/SPI)	CS Position
SPAC13C5.03.1 length_543	OTHER	1.000000	0.000000	
SPAC227.11c.1 length_288	SP	0.412506	0.587495	CS pos: 18-19. Pr: 0.5531
SPAPB24D3.10c.1 length_969	SP	0.000231	0.999745	CS pos: 24-25. Pr: 0.9680
```

**Comment by ValWood** [comment]
SPAC13C5.03 signal_peptide
SPAC227.11c signal_peptide
SPAPB24D3.10c signal_peptide

I would expect all 3 of these to have one.
SPAC13C5.03 tht1 is in the inner NM and I think these get there via the ER......

Need to check  the aa difference in the first 30? tht1 is not listed as being changed. Maybe we have the incorrect gene structure?

**Comment by ValWood** [comment]
does this have one?
>tht1   - 427482: 429318  MW: 64390.7
MKFHPTRPLESFMKYSNVAFSEGLAGFDSLAHVYQALLKKSTCYQEVAATLISKCSLLNT
ELTIDNRIHSAIQMTLCDFERSQILAPSECVRGSQSECVSKLESTSTWWLSFTSHFHDVN
HLCRLANLEMQKELSIEVNMNVTLVQKQFLEMVILHLRNFESVTDKMNQRIDKFDGKFNS
VIENSFKDINFRVNQEIMGLVELQNHQQEGMVQQKEILSTIKQLKSEIFDINSFFANFIE
ESAGYSNSLIEKLNEKFTSENAIALSAIGKYTSEFSAFMEKRIKNLITTTEDSLQQSVQS
NIDFVNSGFQPLYDLTIQLKEELQSLKRLSSEQQNLQHEQILQWKSDFLNVSKDHLKVLQ
QLRPLIDIVEKFMNVYFKGLSNIISSFAFIGFTLFATLSSLFFKVLKIHRRPIIVFGSLS
IIFIHIYCFKITSWVNLYGWITCTIARTLSFIKLNIRTFYLTAFLCALLNFLRYLKYRNS
KKDTELSLFLPAPEECNIYHNEHIQVQEDNYLCPIENSLIDLFGSENNKEKLGKQENVRF
AFLNSESLEQSPWWD

**Comment by kimrutherford** [comment]
> does this have one?

I pasted it in here: https://services.healthtech.dtu.dk/services/SignalP-6.0/

It predicted no signal peptide.

**Comment by kimrutherford** [comment]
> I think we should say "binding sites imported from UniProt"

That change is live now.

>>   So I guess we know it's safe to show the features for genes where our protein sequence matches UniProt's sequence?
>
> Yes, I was thinking that. We will lose a lot but when the sequence updates filter through to UniProtKB they will come through.

The code now checks the sequence from UniProt.  If it matches the binding sites, active sites, beta strand and helices for that protein are included in the protein viewer.

The sequences that don't match are written to the `.web-json-write` log file:
https://curation.pombase.org/dumps/builds/pombase-build-2024-08-05/logs/log.2024-08-04-21-04-31.web-json-write

The list is here too: https://www.pombase.org/results/from/id/4454154c-4b10-4ca6-9121-3ca0b3270536

For now the signal peptide and transit peptide locations are show even if the sequence doesn't match.  Perhaps we should remove those too?

**Comment by kimrutherford** [comment]
> In case anything looks interesting, here are the options for extra fields when downloading from the UniProt advanced search:

I noticed "Turn" under Beta strand and Helix.  Would that track be useful?

An example is here: https://www.uniprot.org/uniprotkb/Q9Y823/feature-viewer

**Comment by kimrutherford** [comment]
> So was that just these 7?

In that case I was just checking the features imported from UniProt, which was easy to do.

I've now made a list of the genes where the UniProt sequence differs in the first 40 amino acids:
https://www.pombase.org/results/from/id/0ef3b3e4-751f-45bf-af8f-07f1fa9fd5c3

When I intersect that list against the full list of signal peptides, we get 9:
https://www.pombase.org/results/from/id/c27e1da0-3a1c-49fe-8edf-f63f4ff2e348

And against the list of transit peptides there are 6:
https://www.pombase.org/results/from/id/c6552aaa-baec-49ef-b05c-118b84f818d7

**Comment by kimrutherford** [comment]
> I noticed "Turn" under Beta strand and Helix. Would that track be useful?

That's done now.  I'll check on Wednesday morning.

**Comment by kimrutherford** [comment]
> When I intersect that list against the full list of signal peptides, we get 9:
> https://www.pombase.org/results/from/id/c27e1da0-3a1c-49fe-8edf-f63f4ff2e348

I put these 9 sequences into SignalP.  It predicts signal peptides for 3 of them:

 - SPBC1347.05c
 - SPAPB24D3.10c
 - SPAC227.11c  (lower probability  ~0.58)

I'll remove the signal peptide annotations for these genes from `pombe-embl/supporting_files/manual_so_term_annotations.tsv`:
 - SPAC13C5.03
 - SPAC1F12.09
 - SPBC13E7.05
 - SPBC30D10.09c
 - SPBC337.16
 - SPCC622.11

**Comment by ValWood** [comment]
These should have signal peptiides though:

> I'll remove the signal peptide annotations for these genes from pombe-embl/supporting_files/manual_so_term_annotations.tsv:

SPAC13C5.03
SPAC1F12.09
SPBC13E7.05
SPBC30D10.09c
SPBC337.16
SPCC622.11

**Comment by kimrutherford** [comment]
> These should have signal peptiides though:

I managed to install Phobius on my desktop.  I checked the 9 proteins with Phobius.  The predictions matched SignalP.

**Comment by ValWood** [comment]
SignalP:

One of the most widely used tools for signal peptide prediction.
SignalP 5.0, the latest version, utilizes deep learning techniques to improve accuracy.
It can differentiate between three types of signal peptides: Sec/SPI (bacterial), Sec/SPII (lipoprotein), and Tat/SPI (twin-arginine translocation).
Phobius:

Integrates signal peptide prediction with topology prediction.
Useful for distinguishing between signal peptides and transmembrane regions.
DeepSig:

Uses deep learning methods to predict signal peptides.
Claims to offer improved performance over traditional methods like SignalP.


Maybe we should also try DeepSig?

**Comment by kimrutherford** [comment]
> Maybe we should also try DeepSig?

I just tried and got the same result:

![image](https://github.com/user-attachments/assets/35343180-d4bf-4227-9675-e1653274601b)

**Comment by kimrutherford** [comment]
6 genes without signal peptides:
https://www.pombase.org/results/from/id/5fcb9395-8d2f-407e-8ef9-fa6050cf01a2

**Comment by ValWood** [comment]
Review of features

 - Names & Taxonomy
   - Entry Name IGNORE
   - Gene Names IGNORE
   - Gene Names (ordered locus) IGNORE
   - Gene Names (ORF) IGNORE
   - Gene Names (primary) IGNORE 
   - Gene Names (synonym) IGNORE
   - Organism IGNORE
   - Organism (ID) IGNORE
   - Protein names IGNORE
   - Proteomes IGNORE
   - Taxonomic lineage IGNORE
   - Taxonomic lineage (Ids) IGNORE
   - Virus hosts IGNORE
 - Sequences
   - Alternative products (isoforms) IGNORE FOR NOW
   - Alternative sequence IGNORE FOR NOW
   - Erroneous gene model prediction IGNORE 
   - Fragment IGNORE 
   - Gene encoded by IGNORE 
   - Length  IGNORE
   - Mass  IGNORE
   - Mass spectrometry  IGNORE
   - Natural variant  IGNORE
   - Non-adjacent residues  IGNORE
   - Non-standard residue  IGNORE
   - Non-terminal residue  IGNORE
   - Polymorphism  IGNORE
   - RNA Editing  IGNORE
   - Sequence  IGNORE
   - Sequence caution  IGNORE
   - Sequence conflict  IGNORE
   - Sequence uncertainty  IGNORE
   - Sequence version  IGNORE
 - Function  IGNORE
   - Absorption  IGNORE
   - Active site IMPORT
   - Binding site IMPORT
   - Catalytic activity IGNORE
   - Cofactor CAN I SEE A LIST TO CHECK  WE HAVE EVERYTHING
   - DNA binding IGNORE (COVERED BY GO)
   - EC number IGNORE FOR NOW (there are a number of routes to get this, but we do not currently display)
   - Activity regulation IGNOR (GO)
   - Function [CC] IGNORE (GO)
   - Kinetics 
   - Pathway IGNORE (GO)
   - pH dependence IGNORE
   - Redox potential IGNORE 
   - Rhea ID IGNORE (COVERED BY GO)
   - Site ?????
   - Temperature dependence  IGNORE
 - Miscellaneous
   - Annotation  IGNORE
   - Caution  IGNORE
   - Keywords  IGNORE
   - Keyword ID  IGNORE
   - Miscellaneous [CC]  IGNORE
   - Protein existence  IGNORE
   - Reviewed  IGNORE
   - Tools  IGNORE
   - UniParc  IGNORE
   - Comments  IGNORE
   - Features
 - Interaction
   - Interacts with IGNORE FOR NOW
   - Subunit structure  IGNORE FOR NOW
 - Expression IGNORE
   - Developmental stage IGNORE
   - Induction IGNORE
   - Tissue specificity IGNORE
 - Gene Ontology (GO) IGNORE
   - Gene Ontology (biological process) IGNORE
   - Gene Ontology (cellular component) IGNORE
   - Gene Ontology (GO) IGNORE
   - Gene Ontology (molecular function) IGNORE
   - Gene Ontology IDs IGNORE
 - Pathology & Biotech IGNORE
   - Allergenic Properties IGNORE
   - Biotechnological use IGNORE
   - Disruption phenotype  IGNORE FOR NOW
   - Involvement in disease IGNORE 
   - Mutagenesis IGNORE FOR NOW
   - Pharmaceutical use IGNORE
   - Toxic dose IGNORE
 - Subcellular location IGNORE (COVERED BY GO)
   - Intramembrane (COVERED BY GO)
   - Subcellular location [CC] (COVERED BY GO)
   - Topological domain
   - Transmembrane  IMPORTED?
 - PTM / Processing
   - Chain IMPORTED? (Not very useful)
   - Cross-link IMPORTED
   - Disulfide bond IMPRORTED
   - Glycosylation IMPORTED
   - Initiator methionine IGNORE
   - Lipidation IMPORTED???
   - Modified residue  What doe s this provide??? how does it differ from Post-translational modification
   - Peptide IGNORE
   - Post-translational modification What doe s this provide??? how does it differ from Modified residue
   - Propeptide IMPORTED???
   - Signal peptide IMPORTED
   - Transit peptide IMPORTED
 - Structure IGNORE
   - 3D IGNORE IGNORE
   - Beta strand IMPORTED
   - Helix IMPROTED
   - Turn IMPORTED
 - Publications IGNORE
   - PubMed ID IGNORE
   - DOI ID IGNORE
 - Date of IGNORE
   - Date of creation IGNORE
   - Date of last modification IGNORE
   - Date of last sequence modification IGNORE
   - Entry version IGNORE
 - Family & Domains IGNORE
   - Coiled coil  IGNORE ????? ( we get this from INterPRO)
   - Compositional bias
   - Domain [CC] IGNORE ( we get this from INterPRO)
   - Domain [FT] IGNORE ( we get this from INterPRO)
   - Motif IGNORE  ( we get this from INterPRO)
   - Protein families IGNORE ( we get this from INterPRO)
   - Region IGNORE 
   - Repeat IGNORE 
   - Sequence similarities IGNORE 
   - Zinc finger IGNORE

**Comment by ValWood** [comment]
I think this si the current situation, can you confirm and answer my queries about the bold ones?

**Comment by ValWood** [comment]
I'm going to close this. Everything is done but I have one question for some data we could import.
I'm not sure what it covers though. I will open a new ticket about that.

**Comment by ValWood** [comment]
final:
https://github.com/pombase/pombase-chado/issues/1213

**Comment by kimrutherford** [comment]
> Site ?????. what is this????

I'm not sure.  :-)
It seems a bit miscellaneous.  Here are some examples:

```
PAC7D4.07c;    trx2    SITE 24; /note="Deprotonates C-terminal active site Cys"; /evidence="ECO:0000250"; SITE 31; /note="Contributes to redox potential value"; /evidence="ECO:0000250"; SITE 32; /note="Contributes to redox potential value"; /evidence="ECO:0000250"
SPAC4F10.16c;           SITE 548; /note="Involved in the release of the transported lipid into the cytosolic leaflet"; /evidence="ECO:0000250|UniProtKB:C7EXK4"
SPBC530.12c;    yhc1    SITE 373; /note="Crucial for post-translational processing"; /evidence="ECO:0000250"
SPBC32F12.01c;          SITE 181; /note="Important for substrate recognition"; /evidence="ECO:0000250"
SPBC1709.11c;           SITE 250; /note="Histone H3K4me3 binding"; /evidence="ECO:0000250|UniProtKB:Q9UK53"; SITE 261; /note="Histone H3K4me3 binding"; /evidence="ECO:0000250|UniProtKB:Q9UK53"; SITE 265; /note="Histone H3K4me3 binding"; /evidence="ECO:0000250|UniProtKB:Q9UK53"; SITE 273; /note="Histone H3K4me3 binding"; /evidence="ECO:0000250|UniProtKB:Q9UK53"
SPBC1734.11;            SITE 329; /note="Involved in dimerization"; /evidence="ECO:0000250"
SPCC18.09c;             SITE 41; /note="Interaction with DNA"; /evidence="ECO:0000269|PubMed:21984210, ECO:0000269|PubMed:26007660"
SPCC330.10;             SITE 121; /note="mRNA cap binding"; /evidence="ECO:0000255|PROSITE-ProRule:PRU00895"; SITE 127; /note="mRNA cap binding"; /evidence="ECO:0000255|PROSITE-ProRule:PRU00895"; SITE 152; /note="mRNA cap binding"; /evidence="ECO:0000255|PROSITE-ProRule:PRU00895"; SITE 195; /note="mRNA cap binding"; /evidence="ECO:0000255|PROSITE-ProRule:PRU00895"; SITE 283; /note="mRNA cap binding"; /evidence="ECO:0000255|PROSITE-ProRule:PRU00895"; SITE 350; /note="mRNA cap binding"; /evidence="ECO:0000255|PROSITE-ProRule:PRU00895"
SPCC1672.10;    hat2    SITE 296; /note="Important for interaction with HAT1"; /evidence="ECO:0000250|UniProtKB:P39984"
SPBP8B7.24c;            SITE 116..117; /note="Cleavage; by atg4"; /evidence="ECO:0000250|UniProtKB:P38182"
SPBC887.12;             SITE 466; /note="Involved in the release of the transported lipid into the cytosolic leaflet"; /evidence="ECO:0000250|UniProtKB:C7EXK4"
SPAC637.12c;    esa1 kat5       SITE 321; /note="Important for catalytic activity"; /evidence="ECO:0000250|UniProtKB:Q08649"
```

**Comment by kimrutherford** [comment]
> Cofactor CAN I SEE A LIST TO CHECK WE HAVE EVERYTHING

[uniprotkb_organism_id_284812_2024_09_04-cofactor.tsv.txt](https://github.com/user-attachments/files/16869436/uniprotkb_organism_id_284812_2024_09_04-cofactor.tsv.txt)

**Comment by ValWood** [comment]
OK we can ignore SITE

**Comment by ValWood** [comment]
OK I checked the first 20 or so cofactor and all are present as IEA GO binding annotations (I expected they would be but I wanted to check). Any where the coordinates are known have binding site annotations i nthe protein viewer (so we can ignore cofactor)

**Comment by kimrutherford** [comment]
> Lipidation IMPORTED???

We're not loading the Lipidation data.  Here's a sample of what it looks like.  There are only 51 genes from UniProt that have this field:

```
 LIPID 485; /note="GPI-anchor amidated serine"; /evidence="ECO:0000255"                                                                     
 LIPID 202; /note="S-geranylgeranyl cysteine"; /evidence="ECO:0000250|UniProtKB:P62745"
 LIPID 404; /note="S-farnesyl cysteine"; /evidence="ECO:0000250"
 LIPID 116; /note="Phosphatidylethanolamine amidated glycine"; /evidence="ECO:0000250|UniProtKB:P38182"
 LIPID 182; /note="S-farnesyl cysteine"; /evidence="ECO:0000250"
 LIPID 203; /note="S-geranylgeranyl cysteine"; /evidence="ECO:0000250|UniProtKB:P36586"; LIPID 205; /note="S-geranylgeranyl cysteine"; /evi…
 LIPID 216; /note="S-farnesyl cysteine"; /evidence="ECO:0000250"
 LIPID 202; /note="S-geranylgeranyl cysteine"; /evidence="ECO:0000269|PubMed:1597466"; LIPID 203; /note="S-geranylgeranyl cysteine"; /evide…
 LIPID 199; /note="S-geranylgeranyl cysteine"; /evidence="ECO:0000250"; LIPID 201; /note="S-geranylgeranyl cysteine"; /evidence="ECO:000025…
 LIPID 213; /note="S-geranylgeranyl cysteine"; /evidence="ECO:0000269|PubMed:1597466"; LIPID 214; /note="S-geranylgeranyl cysteine"; /evide…
 LIPID 2; /note="N-myristoyl glycine"; /evidence="ECO:0000250"; LIPID 3; /note="S-palmitoyl cysteine"; /evidence="ECO:0000250"
 LIPID 189; /note="S-geranylgeranyl cysteine"; /evidence="ECO:0000250"
 LIPID 2; /note="N-myristoyl glycine"; /evidence="ECO:0000269|PubMed:14722091"
 LIPID 199; /note="S-geranylgeranyl cysteine"; /evidence="ECO:0000250"
 LIPID 197; /note="S-geranylgeranyl cysteine"; /evidence="ECO:0000250"
 LIPID 200; /note="S-geranylgeranyl cysteine"; /evidence="ECO:0000250"
 LIPID 199; /note="GPI-anchor amidated serine"; /evidence="ECO:0000255"
 LIPID 206; /note="S-geranylgeranyl cysteine"; /evidence="ECO:0000250|UniProtKB:P36586"; LIPID 208; /note="S-geranylgeranyl cysteine"; /evi…
 LIPID 516; /note="GPI-anchor amidated serine"; /evidence="ECO:0000255"
```

**Comment by kimrutherford** [comment]
> - Modified residue What does this provide??? how does it differ from Post-translational modification
> - Post-translational modification What does this provide??? how does it differ from Modified residue

Here is a sample of the "Modified residue" data:

```                                                                                                                                                                    
 SPAC144.13c;   │ MOD_RES 62; /note="Phosphoserine"; /evidence="ECO:0000269|PubMed:10921878"
 SPBC428.11;    │ MOD_RES 210; /note="N6-(pyridoxal phosphate)lysine"; /evidence="ECO:0000250|UniProtKB:P06721"
 SPAC22A12.07c; │ MOD_RES 451; /note="Phosphothreonine"; /evidence="ECO:0000269|PubMed:18257517"
 SPAC23C4.08;   │ MOD_RES 202; /note="Cysteine methyl ester"; /evidence="ECO:0000250|UniProtKB:P62745"
 SPAC2F3.09;    │ MOD_RES 377; /note="N6-(pyridoxal phosphate)lysine"; /evidence="ECO:0000250|UniProtKB:P18079"
 SPAC31G5.15;   │ MOD_RES 912; /note="Pyruvic acid (Ser); by autocatalysis"; /evidence="ECO:0000255|HAMAP-Rule:MF_03209"
 SPBC428.02c;   │ MOD_RES 256; /note="N6-(pyridoxal phosphate)lysine"; /evidence="ECO:0000250"
 SPAC10F6.09c;  │ MOD_RES 105; /note="N6-acetyllysine"; /evidence="ECO:0000250"
```

And here is a sample of the "Post-translational modification" data.  It seems to be free text notes.  Some are quite long:

```
 SPBP16F5.04;            │ PTM: Autoubiquitinated at Cys-90; undergoes 'Lys-48'-linked polyubiquitination, which leads to proteasome-dependent protein degradation. {ECO:0000250|UniProtKB:Q02159}.
 SPAC144.13c;            │ PTM: Phosphorylated by cdc2-cdc13-CDK complex. This targets srw1 for proteolysis which in turn promotes cdc13 turnover. Dephosphorylated during G1 arrest. {ECO:0000269|PubMed:10921878, ECO:0000269|PubMed:18257517}.
 SPAC11E3.13c;           │ PTM: The GPI-anchor is attached to the protein in the endoplasmic reticulum and serves to target the protein to the cell surface. There, the glucosamine-inositol phospholipid moiety is cleaved off and the GPI-modified mannoprotein is covalently attached via its lipidless GPI glycan remnant to the 1,6-beta-glucan of the outer cell wall layer.
 SPAC14C4.09;            │ PTM: Not glycosylated.
 SPAC17A5.04c;           │ PTM: Glycosylated. {ECO:0000269|PubMed:14871934}.
 SPAC17G6.12;            │ PTM: Neddylated; enhancing the ubiquitin-ligase activity. {ECO:0000250|UniProtKB:P47050}.
 SPAC19G12.14;           │ PTM: Phosphorylated by casein kinase I. Phosphorylation inactivates the enzyme. {ECO:0000269|PubMed:9873063}.                                                                                                                                                                                                                                                                                                                                   
 SPAC20G4.03c;           │ PTM: Autophosphorylated.
 SPAC22A12.08c;          │ PTM: [Isoform 1]: Proteolytically cleaved, presumably during its import into the mitochondrion by mitochondrial processing peptidase. {ECO:0000269|PubMed:29958934}.
 SPAC23C4.08;            │ PTM: Palmitoylated by the erf2-erf4 complex. {ECO:0000269|PubMed:23843742}.
 SPCC962.02c;            │ PTM: Phosphorylated by ark1. {ECO:0000269|PubMed:11950927}.
 SPAC31G5.15;            │ PTM: Is synthesized initially as an inactive proenzyme. Formation of the active enzyme involves a self-maturation process in which the active site pyruvoyl group is generated from an internal serine residue via an autocatalytic post-translational modification. Two non-identical subunits are generated from the proenzyme in this reaction, and the pyruvate is formed at the N-terminus of the alpha chain, which is derived from the c…
 SPAC31G5.18c;           │ PTM: The N-terminal UBL (ubiquitin-like) propeptide is cleaved at Gly-84 by the deubiquitinating enzymes ubp5 and ubp15; the resulting mature sde2 associates with spliceosomes. {ECO:0000269|PubMed:28947618, ECO:0000269|PubMed:36095128}.; PTM: Polyubiquitinated; ubiquitination is partially dependent on ubr11. {ECO:0000269|PubMed:28947618}.
 SPAC3A11.08;            │ PTM: Neddylated; enhancing the ubiquitin-ligase activity. {ECO:0000250|UniProtKB:P47050}.
 SPAC56E4.06c;           │ PTM: Cleaved by autocatalysis into a large and a small subunit. {ECO:0000250}.
 SPAC6B12.11;            │ PTM: Phosphorylated by cdc2 at the onset of S-phase. {ECO:0000269|PubMed:11937031, ECO:0000269|PubMed:18257517}.
 SPAC9E9.11;             │ PTM: The N-terminus is blocked. {ECO:0000269|PubMed:10438489}.
 SPBC16E9.18;            │ PTM: Is synthesized initially as an inactive proenzyme. Formation of the active enzyme involves a self-maturation process in which the active site pyruvoyl group is generated from an internal serine residue via an autocatalytic post-translational modification. Two non-identical subunits are generated from the proenzyme in this reaction, and the pyruvate is formed at the N-terminus of the alpha chain, which is derived from the c…
 SPBC30D10.10c;          │ PTM: Phosphorylation at Thr-1972 in the ATP-binding region by AKT1 strongly reduces kinase activity. {ECO:0000269|PubMed:24247430}.

```

**Comment by kimrutherford** [comment]
> Coiled coil IGNORE ????? ( we get this from INterPRO)

I downloaded the coiled coil data from pfam while it was still existed.  InterPro doesn't provide coiled coil data in their XML file.

We also get the low complexity regions and disordered regions from the pfam download.  The file is from 2021 so it's quite out of date now.

**Comment by ValWood** [comment]
I will ask if InterPro could include it in their XML....

**Comment by ValWood** [comment]
We can map all of the:

 LIPID 485; /note="GPI-anchor amidated serine"; /evidence="ECO:0000255"   
maps to  N-seryl-glycosylphosphatidylinositolethanolamine
https://www.pombase.org/term/MOD:00171
(currently 1 in PomBase)

LIPID 202; /note="S-geranylgeranyl cysteine"; /evidence="ECO:0000250|UniProtKB:P62745"
maps to  geranylgeranylated residue (All should be cysteine)
https://www.pombase.org/term/MOD:00441
(currently 2 in PomBase)

LIPID 404; /note="S-farnesyl cysteine"; /evidence="ECO:0000250"
maps to S-farnesyl-L-cysteine
https://www.pombase.org/term/MOD:00111
(currently 3 in PomBase)

LIPID 116; /note="Phosphatidylethanolamine amidated glycine"; /evidence="ECO:0000250|UniProtKB:P38182"
maps to N-glycyl-1-(phosphatidyl)ethanolamine
https://www.pombase.org/term/MOD:00351
(currently 1 in PomBase)

**Comment by ValWood** [comment]
LIPID 2; /note="N-myristoyl glycine"; /evidence="ECO:0000269|PubMed:14722091"
to N-myristoylglycine 
https://www.pombase.org/term/MOD:00068
(currently 21 in PomBase)

What does that leave?

**Comment by ValWood** [comment]
> I will ask if InterPro could include it in their XML....

I mailed Interpro to ask if this is possible, but now I started worrying that we have a lot of features from intoPro that   incorrect coordinates based on the current Pase sequences  (because they are based on UniProt) i.e. everything that has any features does have coordinate changes. 
This will include InterPro domains but also the coil-coil etc. How much hassle would it be, instead of updating with every InterPro release,  to run InterProScan locally a couple of times a year ?
I would rather the features were accurate coordinates, but maybe lagging a little bit behind InterPro in content ( because there is never too much new stuff for pombe in a release)

**Comment by kimrutherford** [comment]
> How much hassle would it be, instead of updating with every InterPro release, to run InterProScan locally a couple of times a year ?

Last time I tried to install InterProScan it was too hard and I failed.  That was a long time ago though and they now provide a helpful Docker image.  I'll give it a go.  There will be a bit of downstream work because the output format of InterProScan isn't the same as the XML file from InterPro.

**Comment by kimrutherford** [comment]
> What does that leave?

Here's the complete list.  Only 51 genes have this type of data.

[uniprotkb_lipid.tsv.txt](https://github.com/user-attachments/files/16908608/uniprotkb_lipid.tsv.txt)

**Comment by ValWood** [comment]
The remaining mappings



SPBC13G1.11;	LIPID 193; /note="S-palmitoyl cysteine"; /evidence="ECO:0000250"; 
maps to https://www.pombase.org/term/MOD:00115
S-palmitoyl-L-cysteine


SPBPJ4664.02;	LIPID 3944; /note="GPI-anchor amidated alanine"; /evidence="ECO:0000255"0250"
https://www.pombase.org/term/MOD:00818
glycosylphosphatidylinositolated residue
(MOD does not have further specificity)

SPAC212.08c;	LIPID 96; /note="GPI-anchor amidated glycine"; /evidence="ECO:0000255"
https://www.pombase.org/term/MOD:00818
glycosylphosphatidylinositolated residue
(MOD does not have further specificity)

SPCC1322.10;	LIPID 242; /note="GPI-like-anchor amidated asparagine"; /evidence="ECO:0000255"
Ignore

**Comment by kimrutherford** [comment]
I've moved the Lipidation stuff to:
   - pombase/pombase-chado#1220

And the InterPro task is here:
 - pombase/pombase-chado#1218

So I think we can close this.
